specs/IFrameMemoryUsage.md

```diff
@@ -0,0 +1,306 @@
+IFrame Memory Usage API
+===
+
+# Background
+The WebView2 team has been asked to provide support for monitoring iframe 
+memory usage. The end developer needs to know how much memory each iframe 

```

Comment:
This API title is misleading, because it in fact is not an API for getting IFrame memory usage at all.

Rather, it's an API for getting the process IDs for a WebView2 and its frames. One thing you could do with the process ID is use it to get memory usage information, but that's not part of the API being presented.

Furthermore, this API also gives you the process ID for the WebView2 itself, not just for the frames.

Note also that there doesn't seem to be a way to query how much memory is being used by a specific frame. You can get total the memory usage for all frames in a single render process (which is all the frames from the same origin, presumably), but you don't know how much of that memory is dedicated to each frame. You also have to be careful not to double-count the memory if you have multiple frames in the same render process.

Reply:
This API initially asked by a partner team who wants to monitor iframe memory usage. Later it extends to include WebView2's main frame. Agree that process IDs can not directly bind with memory usage. 

For iframes share one render process case, there's no appropriate way to get exact how much memory each iframe use.  The partner team who raises this requirement aware of this and has some strategies to mitigate. But for general public, I agree this API won't able to provide accurate memory usage for each iframe/main frame.

Reply:
Please update title and references throughout.